README.md

```diff
@@ -1,3 +1,42 @@
+# Review
+
+## Markdow
+
+under 'Review' titles.
+
+## Rust
+
+in comment containing `[Review]`
+
+## changes
+
+I also did propose some simple changes:
+- minor syntactic changes.
+- additional comment when it would have save me code reading (I did not try to document all, just what would have help me moving through code while reading).
+
+## Global question
+
+### Value key stoarge.
+store whole 32 byte
+
+### Key type
+
+- index.rs: we only need 8 start byte so most size fits.
+
+### Index
+
+Chunk (page) size to 512 byte (64 entries):
+I am not sure if the choice here is related to the page size or the number of entries
+(with faster search would it be good to use bigger page size).
+
+### Hash
+
+Not really sure why we use salt option.
```

Comment:
That's to mitigate an attack has been demonstrated in the audit. Substrate user may brute force finding such runtime storage keys, that result in the database keys sharing bits required to overflow the chunk quickly, which leads to index size blowing up. Salting makes such an attack against the whole network impossible.